src/appmake/zx.c

```diff
@@ -155,6 +156,7 @@ option_t zx_options[] = {
     { 0 , "merge",     "Merge a custom loader from external TAP file", OPT_STR, &zxt.merge },
     { 0 , "blockname", "Name of the code block in tap file", OPT_STR, &zxt.blockname },
     { 0,  "clearaddr", "Address to CLEAR at",       OPT_INT,   &zxt.clear_address },
+    { 0,  "bank-order","Order banks this way in the tap file (default '01234567')", OPT_STR, &zxt.bank_order },
```

Comment:
Unfortunately this change is incompatible with the built in loader here: https://github.com/z88dk/z88dk/blob/master/lib/target/zx/classic/zx_tapeloader.asm

Reply:
Although this is incompatible with the existing loader I do think it is a useful feature to change the order the banks are written to the tap file. i.e. loading bank 5 first which could contains the screen memory plus user code negates the need to use the --screen option. How would you feel about an new loader where the user passes a pointer to a table with the banks to be loaded? This way the user can specify which order the banks are loaded? No worries if this doesn't work for you, I can back out this change.

Reply:
Hi @suborb , I've been looking at https://github.com/z88dk/z88dk/blob/master/lib/target/zx/classic/zx_tapeloader.asm and I see a few things I don't understand.
For each bank, load_block is called followed by a 'ret c' presumably to stop loading other banks if there is an error. However, load_block does 'and a' before it returns so the carry flag is always reset.
Also, the BASIC variables used (current memory page and ERR-SP) and are in bank 5, so calling load_block after bank 5 has been loaded would clobber these two memory locations in bank 5. Maybe some refactoring is needed here?

Reply:
I found the same effect can be achieved by creating a tap file for each bank and concatenating them together. I have reverted the changes.

Reply:
@iratahack yes, it looks like I wasn't expecting people to load anything into banks 5,2,0.

The system variable thing is tricky - we're using a ROM routine to do the loading - I suspect we don't need to save the bank 23388, but 23613 I think does need to be saved.

I'll sort out the carry flag issue - I was obviously in two minds about what to do there!

Reply:
@suborb, I took a look at the disassembly for ld-bytes (call 0x556) I couldn't see where it uses err-sp (23613) so maybe its not needed. Take a look https://skoolkid.github.io/rom/asm/5C3D.html and https://skoolkid.github.io/rom/asm/0556.html.

Reply:
I thought it was for the break key detection?

It looks like there's a return via SA_LD_RET: https://skoolkid.github.io/rom/asm/053F.html which calls rst $08 with an error.

Reply:
Ah, yes. I got it now. I guess the only way to be 100% sure code isn't clobbered is to copy the ROM loader and remove the break detection and therefore the dependency on  err-sp. Or just ignore err-sp since there is no way to recover from break being pressed.

Reply:
@suborb , I pulled the ROM ld-bytes routine for my own project and removed the break checking, verify code and block header type checking and changed the border colors. It's less than 150 bytes when assembled. If you want to snag it instead of calling the ROM, feel free. It's here https://github.com/iratahack/CastleEscape/blob/develop/src/crt/ld_bytes.asm